src/test/unit_tests/test_bucketspace_versioning.js

```diff
@@ -407,7 +407,52 @@ mocha.describe('bucketspace namespace_fs - versioning', function() {
                 const exist = await version_file_exists(full_path, mpu_key1, '', prev_version_id);
                 assert.ok(exist);
             });
+
+            mocha.it('mpu object - versioning enabled - put object while running mpu. mpu should move to .versions dir and not latest', async function() {
+                const mpu_res = await s3_uid6.createMultipartUpload({ Bucket: bucket_name, Key: mpu_key1 });
+                const upload_id = mpu_res.UploadId;
+                const part1 = await s3_uid6.uploadPart({
+                    Bucket: bucket_name, Key: mpu_key1, Body: body1, UploadId: upload_id, PartNumber: 1 });
+                const res_put_object = await s3_uid6.putObject({Bucket: bucket_name, Key: mpu_key1, Body: body1});
+                const res = await s3_uid6.completeMultipartUpload({
+                    Bucket: bucket_name,
+                    Key: mpu_key1,
+                    UploadId: upload_id,
+                    MultipartUpload: {
+                        Parts: [{
+                            ETag: part1.ETag,
+                            PartNumber: 1
+                        }]
+                    }
+                });
+                const exist = await version_file_exists(full_path, mpu_key1, '', res.VersionId);
+                assert.ok(exist);
+                const comp_res = await compare_version_ids(full_path, mpu_key1, res_put_object.VersionId, res.VersionId);
+                assert.ok(comp_res);
+            });
+
+            mocha.it('mpu object - versioning enabled - put delete marker while running mpu. mpu should move to .versions dir and not latest', async function() {
+                const mpu_res = await s3_uid6.createMultipartUpload({ Bucket: bucket_name, Key: mpu_key1 });
+                const upload_id = mpu_res.UploadId;
+                const part1 = await s3_uid6.uploadPart({
+                    Bucket: bucket_name, Key: mpu_key1, Body: body1, UploadId: upload_id, PartNumber: 1 });
+                await s3_uid6.deleteObject({Bucket: bucket_name, Key: mpu_key1});
+                const res = await s3_uid6.completeMultipartUpload({
+                    Bucket: bucket_name,
+                    Key: mpu_key1,
+                    UploadId: upload_id,
+                    MultipartUpload: {
+                        Parts: [{
+                            ETag: part1.ETag,
+                            PartNumber: 1
+                        }]
+                    }
+                });
+                const exist = await version_file_exists(full_path, mpu_key1, '', res.VersionId);
+                assert.ok(exist);
+            });
```

Comment:
Could you add an assert so we can see that the delete marker is the latest explicitly?

Reply:
in case of delete marker there is no latest version. I added an assert that the latest version doesn't exist